llvm/lib/Transforms/IPO/AlwaysInliner.cpp

```diff
@@ -37,86 +37,74 @@ bool AlwaysInlineImpl(
     function_ref<BlockFrequencyInfo &(Function &)> GetBFI) {
   SmallSetVector<CallBase *, 16> Calls;
   bool Changed = false;
-  SmallVector<Function *, 16> InlinedFunctions;
-  for (Function &F : M) {
-    // When callee coroutine function is inlined into caller coroutine function
-    // before coro-split pass,
-    // coro-early pass can not handle this quiet well.
-    // So we won't inline the coroutine function if it have not been unsplited
+  SmallVector<Function *, 16> InlinedComdatFunctions;
+
+  for (Function &F : make_early_inc_range(M)) {
```

Comment:
add a comment about the fact that we need this since we may delete the function we're visiting

Reply:
I've checked a number of places `make_early_inc_range()` used - found no comments about its functionality. It seems its usage instead of default iterator is quite obvious, what do you think?

Updated the comment below about `InlinedComdatFunctions` usage.

Reply:
I think it's not immediately obvious since the deletion happens further down so I'd write a comment, but won't object if you omit the comment